collectors/htlcs_collector.go

```diff
@@ -202,12 +202,18 @@ func (h *htlcMonitor) processHtlcEvent(event *routerrpc.HtlcEvent) error {
 
 	ts := time.Unix(0, int64(event.TimestampNs))
 
+	return key, ts
+}
+
+// processHtlcEvent processes all the htlc events we consume from our stream.
+func (h *htlcMonitor) processHtlcEvent(event *routerrpc.HtlcEvent) error {
 	switch e := event.Event.(type) {
 	// If we have received a forwarding event, we add it to our map if it
 	// is not already present. We are ok with duplicate events, because
 	// htlcs are sometimes replayed by the switch, but we want to keep our
 	// earliest timestamp for stats.
 	case *routerrpc.HtlcEvent_ForwardEvent:
+		key, ts := getKeyAndTimestamp(event)
```

Comment:
nit: Looks like we could just call this once, at the beginning of the method?

Reply:
The linter complains about that. It classifies the vars key and tx as unused because they aren't used in every single branch of the switch statement. That is why I extracted to a method and then call the method only from the branches where it's used.

Reply:
Oh, I see. I thought maybe the linter complained about the length of the function. But then it makes sense to leave it the way it is now.